src/rest/SearchEndpoint.ts

```diff
@@ -1421,11 +1420,83 @@ function includeReferrer() {
   };
 }
 
+function includeClientId() {
+  return function(target: Object, key: string, descriptor: TypedPropertyDescriptor<any>) {
+    const { originalMethod, nbParams } = decoratorSetup(target, key, descriptor);
+    descriptor.value = function(...args: any[]) {
+      let clientId = AnalyticsUtils.clientId;
+      if (clientId == null) {
+        clientId = '';
+      }
+
+      if (args[nbParams - 1]) {
+        args[nbParams - 1].requestData.clientId = clientId;
+      } else {
+        const endpointCallParams = _.extend(defaultDecoratorEndpointCallParameters(), {
+          requestData: { clientId: clientId }
+        });
+        args[nbParams - 1] = endpointCallParams;
+      }
+      return originalMethod.apply(this, args);
+    };
+
+    return descriptor;
+  };
+}
+
+function includePageId() {
+  return function(target: Object, key: string, descriptor: TypedPropertyDescriptor<any>) {
+    const { originalMethod, nbParams } = decoratorSetup(target, key, descriptor);
+    descriptor.value = function(...args: any[]) {
+      let pageId = AnalyticsUtils.pageId;
+      if (pageId == null) {
+        pageId = '';
+      }
+
+      if (args[nbParams - 1]) {
+        args[nbParams - 1].requestData.pageId = pageId;
+      } else {
+        const endpointCallParams = _.extend(defaultDecoratorEndpointCallParameters(), {
+          requestData: { pageId: pageId }
+        });
+        args[nbParams - 1] = endpointCallParams;
+      }
+      return originalMethod.apply(this, args);
+    };
+
+    return descriptor;
+  };
+}
+
+function includeLocation() {
+  return function(target: Object, key: string, descriptor: TypedPropertyDescriptor<any>) {
+    const { originalMethod, nbParams } = decoratorSetup(target, key, descriptor);
+    descriptor.value = function(...args: any[]) {
+      let location = AnalyticsUtils.location;
+      if (location == null) {
+        location = '';
+      }
+
+      if (args[nbParams - 1]) {
+        args[nbParams - 1].requestData.location = location;
+      } else {
+        const endpointCallParams = _.extend(defaultDecoratorEndpointCallParameters(), {
+          requestData: { location: location }
+        });
+        args[nbParams - 1] = endpointCallParams;
+      }
+      return originalMethod.apply(this, args);
+    };
+
+    return descriptor;
+  };
+}
+
 function includeVisitorId() {
   return function(target: Object, key: string, descriptor: TypedPropertyDescriptor<any>) {
     const { originalMethod, nbParams } = decoratorSetup(target, key, descriptor);
     descriptor.value = function(...args: any[]) {
-      let visitorId = Cookie.get('visitorId');
+      let visitorId = AnalyticsUtils.visitorId;
```

Comment:
Even when analytics are disabled, a cookie can be left over from previously visiting a page with analytics enabled.

Reply:
Nice fix 👍